### ISSUE TITLE
[Regression] Database doesn't update changed tags

### ISSUE THREAD

**Issue by litemotiv** [Taiko2k/Tauon#1221]
This appears to occur since the 7.8.2 flatpak update on my system.

Steps to reproduce:

**1) This album is missing track numbers**

![2024-09-01_09-07](https://github.com/user-attachments/assets/f7763475-61b1-4b05-ba01-aa86465d3802)

**2) Edit tags in external tag editor**

<img src="https://github.com/user-attachments/assets/96a9d7d2-ed2e-4acf-a114-7b80ada52171" width="250">

Rescanning the files in Tauon doesn't update the track number.

**3) Renaming folder does show the changed tags**

<img src="https://github.com/user-attachments/assets/ac9288c3-f2c4-408d-8049-03d9b9c9e638" width="500">

![2024-09-01_09-15](https://github.com/user-attachments/assets/e2b17ccf-eaeb-48ef-9dd8-f6ac41d4f4a5)

**4) Setting folder name back to original loses the changes again** 

![2024-09-01_09-07](https://github.com/user-attachments/assets/f7763475-61b1-4b05-ba01-aa86465d3802)

**Comment by Taiko2k** [comment]
When you say file rescan are you doing the "Rescan Tags" function? The "Rescan Folder" function doesn't do a tag rescan. 

Otherwise, what file format are the tracks and by what method did you import them?

**Comment by litemotiv** [comment]
> When you say file rescan are you doing the "Rescan Tags" function? The "Rescan Folder" function doesn't do a tag rescan.
> 
> Otherwise, what file format are the tracks and by what method did you import them?

That's interesting, i've only used the 'Rescan Folder' option since i started using Tauon (about a month ago), and it always showed the updates that  i made. But the last few days it didn't seem to work, so i reckoned it was due to the flatpak update. 

I've imported all my files (all flac) on the first run to the Default playlist, and have only worked with that since. I'm using the Tree explorer., and for newly added folders i also do a Rescan Folder to make them appear.  

Perhaps there is a difference when making changes to CUE files, which is what i have mostly done in the last weeks, and changes to the embedded tags. Could that be it? That CUE changes are picked up during Rescan Folder but embedded tags not?

**Comment by Taiko2k** [comment]
Yes that does sound about right, CUE sheets are rescanned when reimported. (If this is correct or ideal behavior is another thing but it's the way it works)

**Comment by litemotiv** [comment]
Perhaps ideally the behaviour would be consistent for all content, both CUE files and embedded information. 

Would it be possible to make the Rescan Folder option fully rescan/replace it's contents?

<img src="https://github.com/user-attachments/assets/da2f1de1-17a6-4204-94ef-5d6cf10dcc7a" width="350">
